robot/impl/resource_manager.go

```diff
@@ -537,115 +539,150 @@ func (manager *resourceManager) completeConfig(
 		manager.logger.CDebugw(ctx, "error resolving dependencies", "error", err)
 	}
 
-	resourceNames := manager.resources.ReverseTopologicalSort()
+	levels := manager.resources.ReverseTopologicalSortInLevels()
 	timeout := rutils.GetResourceConfigurationTimeout(manager.logger)
-	for _, resName := range resourceNames {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-		}
-
-		resChan := make(chan struct{}, 1)
-		resName := resName
-		ctxWithTimeout, timeoutCancel := context.WithTimeout(ctx, timeout)
-		defer timeoutCancel()
-
-		cleanup := rutils.SlowStartupLogger(
-			ctx, "Waiting for resource to complete (re)configuration", "resource", resName.String(), manager.logger)
-
-		lr.reconfigureWorkers.Add(1)
-		goutils.PanicCapturingGo(func() {
-			defer func() {
-				cleanup()
-				resChan <- struct{}{}
-				lr.reconfigureWorkers.Done()
-			}()
-			gNode, ok := manager.resources.Node(resName)
-			if !ok || !gNode.NeedsReconfigure() {
-				return
-			}
-			if !(resName.API.IsComponent() || resName.API.IsService()) {
+	for _, resourceNames := range levels {
+		var levelErrG errgroup.Group
```

Comment:
trying out an [errgroup](https://pkg.go.dev/golang.org/x/sync@v0.7.0/errgroup) here to capture when we need to early exit the entire function due to the top level `ctx` completing.

Reply:
added here: https://github.com/viamrobotics/rdk/pull/3707/commits/6e0d616b64cbb72741bb875277f1169da0257a4d

i missed the part about documenting forceSync, will do

Reply:
Sorry! Edited this in place was poor form on my part.

Reply:
oh no worries, your original comment implied that multiple things needed docs so I was prepared for some back and forth

Reply:
added docs on the forceSync param https://github.com/viamrobotics/rdk/pull/3707/commits/845faf7c938d99fe68b65393fe8e923c3b4cc4b3